frame/treasury/src/lib.rs

```diff
@@ -396,10 +396,14 @@ decl_storage! {
 	add_extra_genesis {
 		build(|_config| {
 			// Create Treasury account
-			let _ = T::Currency::make_free_balance_be(
-				&<Module<T, I>>::account_id(),
-				T::Currency::minimum_balance(),
-			);
+			let account_id = <Module<T, I>>::account_id();
+			let min = T::Currency::minimum_balance();
+			if T::Currency::free_balance(&account_id) < min {
```

Comment:
How are you setting the initial treasury balance?

Why not adding a `extra_genesis` parameter that is a `Option<Balance>` and if that is `None` we set the initial balance to `ED` and otherwise to the given value?

Reply:
we set treasury pallet initial balance through genesis ```endowed_accounts```, for example append ```Treasury::account_id()```, if we use a new ```extra_genesis``` parameter to control the value in treasury pallet self, seems it will cause imbalance, we hope the ```Balances::total_issuance()``` could include this initial value.

Reply:
@shawntabrizi WDYT?

Reply:
I think having `Option<Balance>` in treasury config is also good, but this way is fine too